cmd/kubectl-testkube/commands/tests/common.go

```diff
@@ -1,59 +1,204 @@
+
+		if execution.ExecutionResult.IsCompleted() {
+			fmt.Println()
+
+			uiShellGetExecution(id)
+
+			return
+		}
+	}
+
+	uiShellGetExecution(id)
 }
 
-func uiShellTestGetCommandBlock(id string) {
-	ui.ShellCommand(
-		"Use following command to get test execution details",
-		"kubectl testkube tests execution "+id,
-	)
+func newContentFromFlags(cmd *cobra.Command) (content *testkube.TestContent, err error) {
+	var fileContent []byte
 
-	ui.NL()
+	testContentType := cmd.Flag("test-content-type").Value.String()
+	file := cmd.Flag("file").Value.String()
+	uri := cmd.Flag("uri").Value.String()
+	gitUri := cmd.Flag("git-uri").Value.String()
+	gitBranch := cmd.Flag("git-branch").Value.String()
+	gitPath := cmd.Flag("git-path").Value.String()
+	gitUsername := cmd.Flag("git-username").Value.String()
+	gitToken := cmd.Flag("git-token").Value.String()
+
+	// get file content
+	if file != "" {
+		fileContent, err = ioutil.ReadFile(file)
+		if err != nil {
+			return content, fmt.Errorf("reading file "+file+" error: %w", err)
+		}
+	} else if stat, _ := os.Stdin.Stat(); (stat.Mode() & os.ModeCharDevice) == 0 {
+		fileContent, err = ioutil.ReadAll(os.Stdin)
+		if err != nil {
+			return content, fmt.Errorf("reading stdin error: %w", err)
+		}
+	}
+
+	// content is correct when is passed from file, by uri, ur by git repo
+	if len(fileContent) == 0 && uri == "" && gitUri == "" {
+		return content, fmt.Errorf("empty test content, please pass some test content to create test")
+	}
+
+	// detect content type (git-file need to be everrided manually as we don't)
+	// TODO handle git-file somehow
+	if gitUri != "" && testContentType == "" {
+		testContentType = string(testkube.TestContentTypeGitDir)
+	}
+
+	if uri != "" && testContentType == "" {
+		testContentType = string(testkube.TestContentTypeFileURI)
+	}
+
+	if len(fileContent) > 0 {
+		testContentType = string(testkube.TestContentTypeString)
+	}
+
+	var repository *testkube.Repository
+	if gitUri != "" && gitBranch != "" {
+		if testContentType == "" {
+			testContentType = "git-dir"
+		}
+
+		repository = &testkube.Repository{
+			Type_:    "git",
+			Uri:      gitUri,
+			Branch:   gitBranch,
+			Path:     gitPath,
+			Username: gitUsername,
+			Token:    gitToken,
+		}
+	}
+
+	content = &testkube.TestContent{
+		Type_:      testContentType,
+		Data:       string(fileContent),
+		Repository: repository,
+		Uri:        uri,
+	}
+
+	return content, nil
 }
 
-func uiShellTestWatchCommandBlock(id string) {
-	ui.ShellCommand(
-		"Use following command to get test execution details",
-		"kubectl testkube tests watch "+id,
-	)
+func NewUpsertTestOptionsFromFlags(cmd *cobra.Command, test testkube.Test) (options apiclientv1.UpsertTestOptions, err error) {
+	content, err := newContentFromFlags(cmd)
+
+	ui.ExitOnError("creating content from passed parameters", err)
+
+	name := cmd.Flag("name").Value.String()
+	executorType := cmd.Flag("type").Value.String()
+	namespace := cmd.Flag("test-namespace").Value.String()
```

Comment:
Do we introduce test-namespace parameter?

Reply:
We should think about this as we can have 2 namespaces in this context - 
1. testkube namespace 
2. created CRD namespace 

we can add some global `--testkube-namespace` or something

Reply:
@vsukhin I would discuss this first - what direction should we go and extract this to separate task

Reply:
https://github.com/kubeshop/testkube/issues/982